### PR TITLE
adding annotations and makefile additions to support swagger doc gen

### PR DESCRIPTION
**Pull request by jakeschuurmans**
#### What does this PR do

Adds swagger doc generation support. Able to generate updated api docs with `make gen-docs`

### CODE REVIEW

File: docs/docs.go [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+            }
+        }
+    }
+}`
+
+// SwaggerInfo holds exported Swagger Info so clients can modify it
+var SwaggerInfo = &swag.Spec{
+	Version:          "1.0.0",
+	Host:             "",
+	BasePath:         "/api",
```
Comment (joelrebel) [inline docs/docs.go:348]: shouldn't `BasePath` be set to `/api/v1`

Comment (DoctorVin) [reply docs/docs.go:348]: If that’s something we control (I just noticed it’s a `&swag.Spec`), then we likely have control over both version and the path the api is hosted at, which might be best as `/api/v1/docs`?

File: docs/swagger.json [hunk 2]
```diff
@@ -0,0 +1,335 @@
+{
+    "swagger": "2.0",
+    "info": {
+        "description": "Conditions API expose CRUD actions to condition objects on servers",
+        "title": "Condition orchestrator API",
+        "contact": {},
+        "version": "1.0.0"
```
[the remark this answers is not in the document]
Comment (jakeschuurmans) [reply docs/swagger.json:7]: yup

File: main.go [hunk 3]
```diff
@@ -17,6 +17,11 @@ package main
 
 import "github.com/metal-toolbox/conditionorc/cmd"
 
+// @BasePath /api
```
[the remark this answers is not in the document]
Comment (jakeschuurmans) [reply main.go:20]: i will try, but swagger can be weird with where stuff needs to be.

File: docs/docs.go (hunk 1 above)
Comment (jakeschuurmans) [reply docs/docs.go:348]: yes

File: main.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (DoctorVin) [reply main.go]: Annoying, but it is what it is. Thanks!

**Review by DoctorVin — APPROVED** [review]
LGTM. I think we'll want to come up with an API versioning policy and set the swag info accordingly, but that doesn't block a merge here, IMO.

**Pull request merged**